mlflow/tracking/fluent.py

```diff
@@ -309,7 +310,8 @@ def _get_experiment_id_from_env():
 def _get_experiment_id():
     return int(_active_experiment_id or
                _get_experiment_id_from_env() or
-               (env.get_env(_AUTODETECT_EXPERIMENT) and
+               ((env.get_env(_AUTODETECT_EXPERIMENT_ENV_VAR, "True").lower() !=
```

Comment:
Open question: Is there a use case for disabling autodetection? If not, it may be easier to remove this environment variable entirely.